src/main/java/codeu/controller/ParsingEngine.java

```diff
@@ -0,0 +1,43 @@
+package codeu.controller;
+
+import codeu.model.data.BotRequest;
+import codeu.model.data.BotStructure;
+import codeu.model.data.Message;
+import codeu.model.data.RoutingEngine;
+
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+	
+
+public class parsingEngine {
+	/* 
+	 * Here, regex is a better way to parse bot commands rather than matching them with some keywords.
+	 * Regex can parse varying bot request according to pattern. 
+	 * For example, the following regex can parse "/time", "/  time" or "/time    " as "time". 
```

Comment:
"/   time" wouldn't work, I think? And I don't think it should. "   /time" would though

Add some discussion about the argument group, something along the lines of:
>This regex looks for a command name as /<command_name>, where command_name is alphanumeric, followed by whitespace-separated 0-N alphanumeric arguments.
> Examples:
> /command
> /command arg1
> /command arg1 arg2 arg3 arg4